crates/fuel-gas-price-algorithm/gas-price-analysis/src/main.rs

```diff
@@ -70,83 +58,141 @@ enum Mode {
-
-    let plot_width = 640 * 2 * 2;
-    let plot_height = 480 * 3;
-
-    let root =
-        BitMapBackend::new(file_path, (plot_width, plot_height)).into_drawing_area();
-    root.fill(&WHITE).unwrap();
-    let (window_one, lower) = root.split_vertically(plot_height / 4);
-    let (window_two, new_lower) = lower.split_vertically(plot_height / 4);
-    let (window_three, window_four) = new_lower.split_vertically(plot_height / 4);
-
-    draw_fullness(&window_one, &fullness, "Fullness");
-
-    draw_bytes_and_cost_per_block(&window_two, &bytes_and_costs, "Bytes Per Block");
-
-    draw_profit(
-        &window_three,
-        &actual_profit,
-        &projected_profit,
-        &pessimistic_costs,
-        &format!("Profit p_comp: {p_comp:?}, d_comp: {d_comp:?}"),
-    );
-    draw_gas_prices(
-        &window_four,
-        &gas_prices,
-        &exec_gas_prices,
-        &da_gas_prices,
-        "Gas Prices",
+    // Max actual profit
+    let (index, max_actual_profit) = actual_profit
+        .iter()
+        .enumerate()
+        .max_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *max_actual_profit as f64 / (10_f64).powf(18.);
+    println!("max actual profit: {} ETH at {}", eth, index);
+
+    // Max projected profit
+    let (index, max_projected_profit) = projected_profit
+        .iter()
+        .enumerate()
+        .max_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *max_projected_profit as f64 / (10_f64).powf(18.);
+    println!("max projected profit: {} ETH at {}", eth, index);
+
+    // Min actual profit
+    let (index, min_actual_profit) = actual_profit
+        .iter()
+        .enumerate()
+        .min_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *min_actual_profit as f64 / (10_f64).powf(18.);
+    println!("min actual profit: {} ETH at {}", eth, index);
+
+    // Min projected profit
+    let (index, min_projected_profit) = projected_profit
+        .iter()
+        .enumerate()
+        .min_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *min_projected_profit as f64 / (10_f64).powf(18.);
+    println!("min projected profit: {} ETH at {}", eth, index);
+
+    // Max DA Gas Price
+    let (index, max_da_gas_price) = da_gas_prices
+        .iter()
+        .enumerate()
+        .max_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *max_da_gas_price as f64 / (10_f64).powf(18.);
+    println!(
+        "max DA gas price: {} Wei ({} ETH) at {}",
+        max_da_gas_price, eth, index
     );
 
-    root.present().unwrap();
+    // Min Da Gas Price
+    let (index, min_da_gas_price) = da_gas_prices
+        .iter()
+        .enumerate()
+        .min_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *min_da_gas_price as f64 / (10_f64).powf(18.);
+    println!(
+        "min DA gas price: {} Wei ({} ETH) at {}",
+        min_da_gas_price, eth, index
+    );
 }
 
+pub fn pretty<T: ToString>(input: T) -> String {
```

Comment:
```suggestion
pub fn prettify_numbers<T: Display>(input: T) -> String {
```

nit: improved naming and usage of `Display` since everyone probably implements Display and gets ToString as a side-effect, ToString is okay though, not a hard req